### PR TITLE
chore: add check builds for all supported versions of Angular

### LINKED ISSUE #1343: Property 'change' in type 'Toggle' is not assignable to the same property in base type 'Checkbox'.
Typescript error

angular 10
typescript : 3.9.5
carbon: 10.13.0

## Detailed description

```
ERROR in node_modules/carbon-components-angular/toggle/toggle.component.d.ts:96:5 - 
error TS2416: Property 'change' in type 'Toggle' is not assignable to the same property 
in base type 'Checkbox'.
```

### PR DESCRIPTION
**Pull request by cal-smith**
Closes IBM/carbon-components-angular#1522

closes #1343 since it affected one of the tests

Expands the Check build action to also check the library against every Angular version we support (`[7, 8, 9, 10, 11]`). 

TODO:
- [x] add components to each app

We can expand the set of components and tests we run as needed. For now it should be sufficient to run a build with a small set of components.

### CODE REVIEW

File: integration/ng7/angular.json [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+            "exclude": [
+              "**/node_modules/**"
+            ]
+          }
+        }
+      }
+    }
+  },
+  "defaultProject": "ng7"
+}
```
Comment (zvonimirfras) [inline integration/ng7/angular.json:140]:
```suggestion
}

```

File: integration/ng7/e2e/protractor.conf.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    defaultTimeoutInterval: 30000,
+    print: function() {}
+  },
+  onPrepare() {
+    require('ts-node').register({
+      project: require('path').join(__dirname, './tsconfig.e2e.json')
+    });
+    jasmine.getEnv().addReporter(new SpecReporter({ spec: { displayStacktrace: true } }));
+  }
+};
```
Comment (zvonimirfras) [inline integration/ng7/e2e/protractor.conf.js:28]:
```suggestion
};

```

File: integration/ng7/e2e/tsconfig.e2e.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    "outDir": "../out-tsc/app",
+    "module": "commonjs",
+    "target": "es5",
+    "types": [
+      "jasmine",
+      "jasminewd2",
+      "node"
+    ]
+  }
+}
```
Comment (zvonimirfras) [inline integration/ng7/e2e/tsconfig.e2e.json:13]:
```suggestion
}

```

File: integration/ng7/src/browserslist [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+# For additional information regarding the format and rule options, please see:
+# https://github.com/browserslist/browserslist#queries
+#
+# For IE 9-11 support, please remove 'not' from the last line of the file and adjust as needed
+
+> 0.5%
+last 2 versions
+Firefox ESR
+not dead
+not IE 9-11
```
Comment (zvonimirfras) [inline integration/ng7/src/browserslist:11]:
```suggestion
not IE 9-11

```

File: integration/ng8/angular.json [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+          "configurations": {
+            "production": {
+              "devServerTarget": "ng8:serve:production"
+            }
+          }
+        }
+      }
+    }},
+  "defaultProject": "ng8"
+}
```
Comment (zvonimirfras) [inline integration/ng8/angular.json:129]:
```suggestion
}

```

File: integration/ng8/browserslist [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+# https://github.com/browserslist/browserslist#queries
+
+# You can see what browsers were selected by your queries by running:
+#   npx browserslist
+
+> 0.5%
+last 2 versions
+Firefox ESR
+not dead
+not IE 9-11 # For IE 9-11 support, remove 'not'.
```
Comment (zvonimirfras) [inline integration/ng8/browserslist:12]:
```suggestion
not IE 9-11 # For IE 9-11 support, remove 'not'.

```

File: integration/ng8/e2e/protractor.conf.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    defaultTimeoutInterval: 30000,
+    print: function() {}
+  },
+  onPrepare() {
+    require('ts-node').register({
+      project: require('path').join(__dirname, './tsconfig.json')
+    });
+    jasmine.getEnv().addReporter(new SpecReporter({ spec: { displayStacktrace: true } }));
+  }
+};
```
Comment (zvonimirfras) [inline integration/ng8/e2e/protractor.conf.js:32]:
```suggestion
};

```

File: integration/ng8/tslint.json [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    "no-outputs-metadata-property": true,
+    "template-banana-in-box": true,
+    "template-no-negated-async": true,
+    "use-lifecycle-interface": true,
+    "use-pipe-transform-interface": true
+  },
+  "rulesDirectory": [
+    "codelyzer"
+  ]
+}
```
Comment (zvonimirfras) [inline integration/ng8/tslint.json:91]:
```suggestion
}

```

File: integration/ng9/browserslist [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+# https://github.com/browserslist/browserslist#queries
+
+# You can see what browsers were selected by your queries by running:
+#   npx browserslist
+
+> 0.5%
+last 2 versions
+Firefox ESR
+not dead
+not IE 9-11 # For IE 9-11 support, remove 'not'.
```
Comment (zvonimirfras) [inline integration/ng9/browserslist:12]:
```suggestion
not IE 9-11 # For IE 9-11 support, remove 'not'.

```

File: integration/ng9/e2e/protractor.conf.js [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    defaultTimeoutInterval: 30000,
+    print: function() {}
+  },
+  onPrepare() {
+    require('ts-node').register({
+      project: require('path').join(__dirname, './tsconfig.json')
+    });
+    jasmine.getEnv().addReporter(new SpecReporter({ spec: { displayStacktrace: true } }));
+  }
+};
```
Comment (zvonimirfras) [inline integration/ng9/e2e/protractor.conf.js:32]:
```suggestion
};

```

File: integration/ng9/tslint.json [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+    "no-outputs-metadata-property": true,
+    "template-banana-in-box": true,
+    "template-no-negated-async": true,
+    "use-lifecycle-interface": true,
+    "use-pipe-transform-interface": true
+  },
+  "rulesDirectory": [
+    "codelyzer"
+  ]
+}
```
Comment (zvonimirfras) [inline integration/ng9/tslint.json:148]:
```suggestion
}

```

**Review by zvonimirfras — COMMENTED** [review]
Would it make sense to copy our editor configs and lint options into integrations as well?

Also, fished out all the "missing new line at the end of file"

**Comment by theiliad** [comment]
> Closes #1522
> 
> Expands the Check build action to also check the library against every Angular version we support (`[7, 8, 9, 10, 11]`).
> 
> TODO:
> 
> * [ ]  add components to each app
> 
> We can expand the set of components and tests we run as needed. For now it should be sufficient to run a build with a small set of components.

has the TODO been addressed yet?

File: integration/ng10/.editorconfig [hunk 12]
```diff
@@ -0,0 +1,16 @@
+# Editor configuration, see https://editorconfig.org
+root = true
+
+[*]
+charset = utf-8
+indent_style = space
```
[the remark this answers is not in the document]
Comment (cal-smith) [reply integration/ng10/.editorconfig:6]: Thanks Angular CLI 🤦

**Comment by cal-smith** [comment]
> Would it make sense to copy our editor configs and lint options into integrations as well?

_Probably_ I'll dig into that a bit 🤔 

> Also, fished out all the "missing new line at the end of file"

Thanks 😂 I'll see if there's something automated I can run to fix this stuff

**Comment by cal-smith** [comment]
> > Closes #1522
> > Expands the Check build action to also check the library against every Angular version we support (`[7, 8, 9, 10, 11]`).
> > TODO:
> > 
> > * [ ]   add components to each app
> > 
> > We can expand the set of components and tests we run as needed. For now it should be sufficient to run a build with a small set of components.
> 
> has the TODO been addressed yet?

Yes! I'll check that

**Comment by stanislavgeorgiev** [comment]
The interesting part is according to Angular's team over here https://angular.io/guide/releases, v7 and v8 are no longer supported which begs the question why we're supporting them :)

**Comment by cal-smith** [comment]
> The interesting part is according to Angular's team over here https://angular.io/guide/releases, v7 and v8 are no longer supported which begs the question why we're supporting them :)

Absolutely! Ideally every consuming team would keep up to date with Angular releases, but historically that hasn't been the case. We absolutely encourage every team to move to the latest version whenever possible, but we also try to support the broadest range of Angular versions possible, to give teams time to update.

We also consider _removing_ version support to be a breaking change, so we would need a major version release to drop Angular 7 and 8 - I would expect a new major version to coincide with Carbon v11.

File: integration/ng9/angular.json [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+          }
+        }
+      }
+    }
+  },
+  "defaultProject": "ng9",
+  "cli": {
+    "analytics": false
+  }
+}
```
Comment (zvonimirfras) [inline integration/ng9/angular.json:132]:
```suggestion
}

```

**Review by zvonimirfras — APPROVED** [review]
That one little thing :)

**Review by zvonimirfras — APPROVED** [review]

**Pull request merged**